### PR TITLE
3 folder api & authentication

### PR DESCRIPTION
**Pull request by JanKaczmarkiewicz**
https://github.com/JanKaczmarkiewicz/folder-app/tree/3-folder-api#run-locally

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +24,12 @@ The goal is to create a safe web application for browsing folders.
   - [Branch naming convention](#branch-naming-convention)
   - [Tests](#tests)
 
+## Run locally
```
[the remark this answers is not in the document]
Comment (JanKaczmarkiewicz) [reply README.md:27]: Yeah. Added in  7174f686d7427e3d90544a6d2cfeeab40128fc28

File: backend/src/db/folder.rs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    Directory(Directory),
+    File(File),
+}
+
+pub struct Directory {
+    pub name: &'static str,
+    pub items: &'static [Location],
+}
+
+const ROOT_DIRECTORY: Directory = Directory {
```
[the remark this answers is not in the document]
Comment (JanKaczmarkiewicz) [reply backend/src/db/folder.rs]: fixed in 
* [feat: use real folder structure](https://github.com/JanKaczmarkiewicz/folder-app/pull/4/commits/3cd39146e7dd1d71ea01cad4dde610b44e21a9ab)
* [feat: use real folder structure](https://github.com/JanKaczmarkiewicz/folder-app/pull/4/commits/3cd39146e7dd1d71ea01cad4dde610b44e21a9ab)
* [fix: correct root path](https://github.com/JanKaczmarkiewicz/folder-app/pull/4/commits/9207dfc53991b55119ca6459e11812bfcd6545ad)

File: backend/src/routes/session.rs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        let cookie = Cookie::build(TOKEN, user.id)
+            .secure(true)
+            .same_site(SameSite::None)
+            .finish();
+
+        cookies.add_private(cookie);
+        return Ok(());
+    }
+
+    Err(Status::BadRequest)
```
[the remark this answers is not in the document]
Comment (JanKaczmarkiewicz) [reply backend/src/routes/session.rs]: oversight :) Thanks
* [fix: return Unauthorized status when invalid credentials](https://github.com/JanKaczmarkiewicz/folder-app/pull/4/commits/08cea033ed1918309d7edd89f32440c329413377)
* [test: login negative case](https://github.com/JanKaczmarkiewicz/folder-app/pull/4/commits/c5560a6f0c0e4ea650c537f235f1cc2d777c5a9d)

**Comment by jakule** [comment]
I checked the latest fix and I didn't have any problems.

**Pull request merged**